distributed/scheduler.py

```diff
@@ -3235,6 +3234,17 @@ def worker_objective(self, ts: TaskState, ws: WorkerState) -> tuple:
         else:
             return (start_time, ws.nbytes)
 
+    def worker_objective_no_deps(self, ws: WorkerState) -> tuple[float, float]:
+        """
+        Objective function to determine the least-busy worker.
+
+        Meant for use with tasks where dependencies are not relevant for scheduling
+        (no dependencies, or widely-shared).
+
+        Minimize worker occupancy. If a tie then break with data storage.
+        """
+        return (ws.occupancy / ws.nthreads, ws.nbytes / (ws.memory_limit or 1))
```

Comment:
Happy with not dividing by `memory_limit` if @crusaderky would prefer. See my thoughts in https://github.com/dask/distributed/issues/7266#issuecomment-1308183309.

Reply:
Yes I'd rather use the absolute value

Reply:
Sure. We can keep it consistent for now and then decide what to do everywhere in https://github.com/dask/distributed/issues/7266.

Reply:
The first half of the tuple is sneakily having the same effect, at least as long as amount of available RAM is roughly correlated with amount of threads.
e.g. if you have a 4-threads and a 16-thread worker, and both have occupancy=1, the above algorithm will always prefer the larger worker. To me it would make more sense if it was just `ws.occupancy`, without dividing by number of threads.

Reply:
The truly buggy case I'm fixing here is when `worker-saturation` > 1, and you're picking between workers that still have open threads, and some that don't. You should _always_ pick a worker with an open thread over one that's full, no matter the occupancy.

This might fix the worst-case behavior I'm worried about, but still schedule equally (though not [equitably](https://interactioninstitute.org/wp-content/uploads/2016/01/IISC_EqualityEquity.png)) when there's not much work:

```python
(0 if len(ws.processing) < ws.ntheads else 1, ws.occupancy, ...)
```